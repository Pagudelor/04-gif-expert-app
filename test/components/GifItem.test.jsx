import { render, screen } from "@testing-library/react";
import { GifItem } from "../../src/components/GifItem";

describe('Pruebas en Gif item', ()=> {
  const title= 'adele';
  const url = 'https://adele/';
    test('should match with the snapshot ', () => {
      const {container}= render(<GifItem title= {title} url={url}/>);
      expect( container).toMatchSnapshot();
    });

    test('show display image and url indicated ', () => {
      render(<GifItem title= {title} url={url}/>);
      //screen.debug();
      //expect(screen.getByRole('img').src).toBe(url);
      //expect(screen.getByRole('img').alt).toBe(title);
      const {src, alt}=screen.getByRole('img');
      expect(src).toBe(url);
      expect(alt).toBe(title);
    });

    test('show display component title ', () => {
      render(<GifItem title= {title} url={url}/>);
      expect(screen.getByText ( title )).toBeTruthy();
    });
  });