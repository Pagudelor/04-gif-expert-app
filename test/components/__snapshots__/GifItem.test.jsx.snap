// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en Gif item should match with the snapshot  1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="adele"
      src="https://adele/"
    />
    <p>
       
      adele
    </p>
  </div>
</div>
`;
